File: backend/src/schemas/lk.py
"""LK (личный кабинет) schemas."""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class LkCredentialsCreate(BaseModel):
    """Schema for saving LK credentials."""

    email: str = Field(..., min_length=5, max_length=255, description="LK email")
    password: str = Field(..., min_length=1, max_length=255, description="LK password")


class LkStatusResponse(BaseModel):
    """Schema for LK status response."""

    has_credentials: bool
    last_sync_at: datetime | None


class SessionGradeResponse(BaseModel):
    """Schema for session grade response."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    session_number: str
    subject_name: str
    result: str
    synced_at: datetime


class SemesterDisciplineResponse(BaseModel):
    """Schema for semester discipline response."""

    model_config = ConfigDict(from_attributes=True)

    id: int
    semester_number: int
    discipline_name: str
    control_form: str
    hours: int
    synced_at: datetime


class LkSyncResponse(BaseModel):
    """Schema for sync operation response."""

    grades_synced: int
    disciplines_synced: int
    last_sync_at: datetime


class LkStudentInfo(BaseModel):
    """Student info extracted from LK."""

    full_name: str | None = None
    group_name: str | None = None
    faculty: str | None = None
    course: int | None = None
    speciality: str | None = None

File: frontend/src/types/lk.ts
/**
 * LK (личный кабинет ОмГУ) types.
 */

/** LK connection status. */
export interface LkStatus {
  has_credentials: boolean;
  last_sync_at: string | null;
}

/** Credentials for LK authentication. */
export interface LkCredentials {
  email: string;
  password: string;
}

/** Session grade from LK. */
export interface SessionGrade {
  id: number;
  session_number: string;
  subject_name: string;
  result: string;
  synced_at: string;
}

/** Semester discipline from LK curriculum. */
export interface SemesterDiscipline {
  id: number;
  semester_number: number;
  discipline_name: string;
  control_form: string;
  hours: number;
  synced_at: string;
}

/** Result of sync operation. */
export interface LkSyncResult {
  grades_synced: number;
  disciplines_synced: number;
  last_sync_at: string;
}

/** Result of import operation to app. */
export interface LkImportResult {
  semesters_created: number;
  semesters_updated: number;
  subjects_created: number;
  subjects_updated: number;
}

/** Student info from LK. */
export interface LkStudentInfo {
  full_name: string | null;
  group_name: string | null;
  faculty: string | null;
  course: number | null;
  speciality: string | null;
}
